addons/mail/static/src/composer/composer.js

```diff
@@ -79,7 +79,8 @@ export class Composer extends Component {
     }
 
     addEmoji(str) {
-        this.state.value += str;
+        let caret_position = this.ref.el.selectionStart;
+        this.state.value = this.state.value.substring(0, caret_position) + str + this.state.value.substring(caret_position, this.state.value.length);
```

Comment:
When adding emoji, it should concatenate as follow:
- before selection start
- replace between selection start-end with emoji
- after selection end

That way it properly handles any case of selection.